### DISCUSSION TITLE
HM-MOD-RPI-MOD mit FT232R(L) findet keine Geräte

### DISCUSSION THREAD

**Discussion by josecosta07** [OpenCCU/OpenCCU#2655] [🇩🇪 Fragen&Antworten] [answered]
### Describe the issue you are experiencing

Folgende Problematik: habe HA auf einem Futro S740 über Home Assistant OS laufen. Das RaspberryMatic Addon habe ich installiert und einen umgebauten USB-Adapter (FT232R(L)) mit dem HM-MOD-RPI-PCB angeschlossen. Der Stick wird erkannt und es erscheinen keine Fehler im Log. Allerdings werden im Anlernmodus keine Geräte gefunden.

Das Kuriose ist: ich habe testweise einen RPi 2B angeschlossen und Raspberrymatic aufgespielt und hier läuft der Stick und findet auch die Geräte. Den Rpi habe ich an genau dieselbe Stelle aufgestellt, wie den Futro. Somit gehe ich davon aus, dass es nicht am Stick liegt bzw. der Stick nicht fehlerhaft ist.

Nun meine Frage: muss ich in HAOSS irgendwie irgendwas mit dem USB-Anschluss machen? Freigeben oder andere Treiber installieren? Der Stick wird in HA gefunden und ist unter Hardware gelistet.

Im RaspberryMatic Addon wird der Stick gefunden und ich kann die Gerätesuche starten. Es werden nach 60 Sekunden aber keine Gerte gefunden.

### Describe the behavior you expected

Es werden keine Geräte gefunden. Der Stick wird erkannt. Im Rpi 2 findet derselbe Stick Geräte.

### Steps to reproduce the issue

1. USB-Stick anschließen 
2. System hochfahren
3. RaspberryMatic Addon starten 
4. Gerätesuche starten
5. Nach 60 Sekunden wird nichts gefunden
...

### What is the version this bug report is based on?

Letzte Addon Version

### Which base platform are you running?

ha-addon (HomeAssistant Add-on)

### Which HomeMatic/homematicIP radio module are you using?

HM-MOD-RPI-PCB

### Anything in the logs that might be useful for us?

```txt
…
```

**Comment by jens-maus** [comment, marked as answer]
> Folgende Problematik: habe HA auf einem Futro S740 über Home Assistant OS laufen. Das RaspberryMatic Addon habe ich installiert und einen umgebauten USB-Adapter (FT232R(L)) mit dem HM-MOD-RPI-PCB angeschlossen. Der Stick wird 

Das (eigene bastellei mit FT232 an HM-MOD-RPI-PCB) ist schlichtweg nicht supported. Die HM-Funkmodule brauchen spezielle Kerneltreiber (generic_raw_uart) um bzgl. Latenzzeiten usw. korrekt zu funktionieren. Auch zeigst du keinerlei Logausgaben, etc. und schreibst nur "letzte Addon Version" 👎 

Verschiebe das daher (weil kein Issue) in den Diskussionsbereich damit sich hier vielleicht ein anderer Bastler dem ganzen annimmt. Ich würde dir stattdessen raten: Besorg dir ein `HmIP-RFUSB` und schließ den per USB an, fertig. Das `HM-MOD-RPI-PCB` ist ohnehin obsolete... Dann siehst du auch so werte wie "Carrier Sense" was vmtl. der Grund dafür sein wird warum zwar dein Bastelstick eingebunden wird aber mit deinem Futro nicht geht. Der wird einfach voll mit Störstrahlung belastet und kann daher einfach keine Geräte anlernen...

**Reply by josecosta07** [reply]
Vielen Dank für die schnelle Rückmeldung. Wäre der hmip-rfusb verfügbar, würde ich mir einen besorgen. Allerdings gibt es derzeit schlichtweg keine, weswegen ich mir was gebastelt habe. Aber danke, dass du dir die Zeit genommen hast mir zu antworten!! Vielleicht kann mir ja jemand im Duskissionsbereich helfen!

**Reply by jens-maus** [reply]
Die Frage ist halt einfach: Als was bzw. wie genau hast du den FT232R angebunden? Ich kann mir schwer vorstellen das dieser out-of-the-box erkannt wird. versuchst du hier ein `HmIP-RFUSB` zu "faken" samt USB IDs?

**Comment by josecosta07** [comment]
Ich habe einen FT232R mit FTP_prog und dem HB-RF-USB Template von alexreinert geflasht. Der Stick wird als HB-RF-USB erkannt.

**Reply by josecosta07** [reply]
Auf dem Rpi2b funktioniert er ja auch ganz normal...

**Reply by jens-maus** [reply]
Na das sind doch dann mal wichtige Zusatzinfos die du leider unterschlagen hattest. Wie gesagt, das `HM-MOD-RPI-PCB` ist alt und kann zusätzlich zum DutyCycle leider **nicht** den CarrierSense Wert angeben was quasi dem "Rauschen" bzw. ein Maß für die Störung in der näheren Umgebung entspricht. Und nun kann es eben schlichtweg sein, das in der anderen Konstellation mit dem Futro dieser bzw. andere Dinge in dieser Umgebung das Funkmodul so dermaßen stören das dieses nicht in der Lage ist die Homematic Geräte zu erreichen. Da hilft nur auf das `RPI-RF-MOD` oder `HmIP-RFUSB` Wechseln, denn diese geben diesen Wert aus.

**Reply by josecosta07** [reply]
Kann ich also ausschließen, dass es nichts mit irgendeinem Treiber zu tun hat??? Sondern eher auf ein Problem mit meinem "Funk-Umfeld" zurückzuführen ist? Hatte es bereits mit zwei Verlängerungskabeln versucht, insgesamt 2 Meter, aber findet trotzdem keine Geräte.
